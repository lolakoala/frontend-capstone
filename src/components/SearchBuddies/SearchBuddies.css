.search-buddies {
  margin-bottom: 100px;
}
