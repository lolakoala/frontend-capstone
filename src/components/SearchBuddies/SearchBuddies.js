import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { DropDownMenu, MenuItem, RaisedButton } from 'material-ui';
import { Redirect } from 'react-router-dom';
import backend from './backend';

class SearchBuddies extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: ''
    };
  }

  componentWillMount() {
    this.props.getAllChallenges();
  }

  componentWillReceiveProps = nextProps => {
    this.setState({ value: nextProps.allChallenges[0] });
  }

  handleChange = (event, index, value) => { console.log('in handlechange'); this.setState({value}); }

  handleSubmit = () => {
    const challenge = this.state.value;

    fetch(`${backend}/api/v1/users?user_challenge=${challenge}`)
      .then(res => res.json())
      .then(res => res.users.map(user => {
        return {
          id: user.id,
          userName: user.user_name,
          // img
          aboutMe: user.user_about,
          city: user.user_location,
          email: user.user_email
        };
      }))
      .then(res => this.props.search(res, 'buddies'))
      .catch(error => { throw error; });
    return <Redirect to="/list/buddySearch" />;
  }

  render() {
    const { allChallenges } = this.props;

    return (
      <div className="search-options">
        <DropDownMenu value={this.state.value} onChange={this.handleChange}>
          {allChallenges.map((challenge, index) => {
            return <MenuItem
              value={challenge}
              primaryText={challenge}
              key={`${challenge}${index}`}
            />;
          })}
        </DropDownMenu>
        <RaisedButton label="Submit" onClick={this.handleSubmit} />
      </div>
    );
  }
}

export default SearchBuddies;

SearchBuddies.propTypes = {
  allChallenges: PropTypes.array,
  getAllChallenges: PropTypes.func,
  search: PropTypes.func,
  currentUser: PropTypes.object
};
