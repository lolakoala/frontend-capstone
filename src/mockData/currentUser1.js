const currentUser1 = {
  email: 'lola@gmail.com',
  city: 'Denver',
  userName: 'lolakoala',
  aboutMe: 'Lola loves doggos.'
};

export default currentUser1;
