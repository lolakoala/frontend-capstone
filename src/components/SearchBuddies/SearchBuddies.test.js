import React from "react";
import SearchBuddies from './SearchBuddies.js';
import { shallow, configure } from "enzyme";
import currentUser1 from '../../mockData/currentUser1.js';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('SearchBuddies', () => {
  const mockFn = jest.fn();
  const allChallenges = ['depression', 'anxiety'];
  const wrapper = shallow(<SearchBuddies
    allChallenges={allChallenges}
    getAllChallenges={mockFn}
    search={mockFn}
    currentUser={currentUser1}
    buddySearch={[]}
  />);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match snap', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call function on mount', () => {
    expect(mockFn).toHaveBeenCalledTimes(1);
  });

  it('should have drop down menu', () => {
    const ddm = wrapper.find('DropDownMenu');

    expect(ddm.length).toEqual(1);
  });

  it('should have one button', () => {
    const button = wrapper.find('RaisedButton');

    expect(button.length).toEqual(1);
  });

  it.skip('should change state on change of drop down menu', () => {
    const ddm = wrapper.find('DropDownMenu');

    ddm.simulate('change', { target: { value: allChallenges[1] } });
    expect(wrapper.state().value).toEqual(allChallenges[1]);
  });
});
