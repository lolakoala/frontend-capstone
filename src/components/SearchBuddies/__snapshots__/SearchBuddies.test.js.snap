// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBuddies should match snap 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBuddies
    allChallenges={
        Array [
            "depression",
            "anxiety",
          ]
    }
    currentUser={
        Object {
            "aboutMe": "Lola loves doggos.",
            "city": "Denver",
            "email": "lola@gmail.com",
            "userName": "lolakoala",
          }
    }
    getAllChallenges={[Function]}
    search={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <DropDownMenu
          anchorOrigin={
                    Object {
                              "horizontal": "left",
                              "vertical": "top",
                            }
          }
          animated={true}
          autoWidth={true}
          disabled={false}
          iconButton={<NavigationArrowDropDown />}
          maxHeight={500}
          multiple={false}
          onChange={[Function]}
          openImmediately={false}
          value=""
>
          <MenuItem
                    anchorOrigin={
                              Object {
                                        "horizontal": "right",
                                        "vertical": "top",
                                      }
                    }
                    checked={false}
                    desktop={false}
                    disabled={false}
                    focusState="none"
                    insetChildren={false}
                    primaryText="depression"
                    targetOrigin={
                              Object {
                                        "horizontal": "left",
                                        "vertical": "top",
                                      }
                    }
                    value="depression"
          />
          <MenuItem
                    anchorOrigin={
                              Object {
                                        "horizontal": "right",
                                        "vertical": "top",
                                      }
                    }
                    checked={false}
                    desktop={false}
                    disabled={false}
                    focusState="none"
                    insetChildren={false}
                    primaryText="anxiety"
                    targetOrigin={
                              Object {
                                        "horizontal": "left",
                                        "vertical": "top",
                                      }
                    }
                    value="anxiety"
          />
</DropDownMenu>,
        <RaisedButton
          disabled={false}
          fullWidth={false}
          label="Submit"
          labelPosition="after"
          onClick={[Function]}
          primary={false}
          secondary={false}
/>,
      ],
      "className": "search-options",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "anchorOrigin": Object {
            "horizontal": "left",
            "vertical": "top",
          },
          "animated": true,
          "autoWidth": true,
          "children": Array [
            <MenuItem
              anchorOrigin={
                            Object {
                                          "horizontal": "right",
                                          "vertical": "top",
                                        }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="depression"
              targetOrigin={
                            Object {
                                          "horizontal": "left",
                                          "vertical": "top",
                                        }
              }
              value="depression"
/>,
            <MenuItem
              anchorOrigin={
                            Object {
                                          "horizontal": "right",
                                          "vertical": "top",
                                        }
              }
              checked={false}
              desktop={false}
              disabled={false}
              focusState="none"
              insetChildren={false}
              primaryText="anxiety"
              targetOrigin={
                            Object {
                                          "horizontal": "left",
                                          "vertical": "top",
                                        }
              }
              value="anxiety"
/>,
          ],
          "disabled": false,
          "iconButton": <NavigationArrowDropDown />,
          "maxHeight": 500,
          "multiple": false,
          "onChange": [Function],
          "openImmediately": false,
          "value": "",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "depression0",
            "nodeType": "class",
            "props": Object {
              "anchorOrigin": Object {
                "horizontal": "right",
                "vertical": "top",
              },
              "checked": false,
              "desktop": false,
              "disabled": false,
              "focusState": "none",
              "insetChildren": false,
              "primaryText": "depression",
              "targetOrigin": Object {
                "horizontal": "left",
                "vertical": "top",
              },
              "value": "depression",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "anxiety1",
            "nodeType": "class",
            "props": Object {
              "anchorOrigin": Object {
                "horizontal": "right",
                "vertical": "top",
              },
              "checked": false,
              "desktop": false,
              "disabled": false,
              "focusState": "none",
              "insetChildren": false,
              "primaryText": "anxiety",
              "targetOrigin": Object {
                "horizontal": "left",
                "vertical": "top",
              },
              "value": "anxiety",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "fullWidth": false,
          "label": "Submit",
          "labelPosition": "after",
          "onClick": [Function],
          "primary": false,
          "secondary": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <DropDownMenu
            anchorOrigin={
                        Object {
                                    "horizontal": "left",
                                    "vertical": "top",
                                  }
            }
            animated={true}
            autoWidth={true}
            disabled={false}
            iconButton={<NavigationArrowDropDown />}
            maxHeight={500}
            multiple={false}
            onChange={[Function]}
            openImmediately={false}
            value=""
>
            <MenuItem
                        anchorOrigin={
                                    Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                        }
                        checked={false}
                        desktop={false}
                        disabled={false}
                        focusState="none"
                        insetChildren={false}
                        primaryText="depression"
                        targetOrigin={
                                    Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                        }
                        value="depression"
            />
            <MenuItem
                        anchorOrigin={
                                    Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                        }
                        checked={false}
                        desktop={false}
                        disabled={false}
                        focusState="none"
                        insetChildren={false}
                        primaryText="anxiety"
                        targetOrigin={
                                    Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                        }
                        value="anxiety"
            />
</DropDownMenu>,
          <RaisedButton
            disabled={false}
            fullWidth={false}
            label="Submit"
            labelPosition="after"
            onClick={[Function]}
            primary={false}
            secondary={false}
/>,
        ],
        "className": "search-options",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "anchorOrigin": Object {
              "horizontal": "left",
              "vertical": "top",
            },
            "animated": true,
            "autoWidth": true,
            "children": Array [
              <MenuItem
                anchorOrigin={
                                Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                primaryText="depression"
                targetOrigin={
                                Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                }
                value="depression"
/>,
              <MenuItem
                anchorOrigin={
                                Object {
                                                "horizontal": "right",
                                                "vertical": "top",
                                              }
                }
                checked={false}
                desktop={false}
                disabled={false}
                focusState="none"
                insetChildren={false}
                primaryText="anxiety"
                targetOrigin={
                                Object {
                                                "horizontal": "left",
                                                "vertical": "top",
                                              }
                }
                value="anxiety"
/>,
            ],
            "disabled": false,
            "iconButton": <NavigationArrowDropDown />,
            "maxHeight": 500,
            "multiple": false,
            "onChange": [Function],
            "openImmediately": false,
            "value": "",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "depression0",
              "nodeType": "class",
              "props": Object {
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "checked": false,
                "desktop": false,
                "disabled": false,
                "focusState": "none",
                "insetChildren": false,
                "primaryText": "depression",
                "targetOrigin": Object {
                  "horizontal": "left",
                  "vertical": "top",
                },
                "value": "depression",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "anxiety1",
              "nodeType": "class",
              "props": Object {
                "anchorOrigin": Object {
                  "horizontal": "right",
                  "vertical": "top",
                },
                "checked": false,
                "desktop": false,
                "disabled": false,
                "focusState": "none",
                "insetChildren": false,
                "primaryText": "anxiety",
                "targetOrigin": Object {
                  "horizontal": "left",
                  "vertical": "top",
                },
                "value": "anxiety",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "fullWidth": false,
            "label": "Submit",
            "labelPosition": "after",
            "onClick": [Function],
            "primary": false,
            "secondary": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
